### DISCUSSION TITLE
Taxonomy in localized search results does not get translated

### DISCUSSION THREAD

**Discussion by Benjime** [microsoft-search/pnp-modern-search#2742] [Q&A]
**Version used**
Version 4.8

**Bug description**
In my Search Results webpart, despite the "enable localization" parameter is enabled, and I've mapped to the taxId crawled property like stated in the documentation, the ID is displayed and not the translated value. 

**To Reproduce**
I'm in a multilingual environment (English, German). 
I have a custom taxonomy column (CustBenEcole), linked to a taxonomy termset where the values are translated in three languages
The column is added to a library where a document is present and indexed
On the tenant level, I've used the "RefinableString125" managed property and attached the "ows_taxId_CustBenEcole" crawled property. 
On my search center, in debug mode, the value is well returned : 
_"RefinableString125": "GP0|#eeff7dc1-c6dd-48f4-aaf8-93f4fe0fb731;L0|#0eeff7dc1-c6dd-48f4-aaf8-93f4fe0fb731|HEI - Filière Technologies du vivant;GTSet|#ac166fd5-7dc7-43f2-8bc3-68be0a3cd2e9",_
In my "search filters" webpart, the translation on RefinableString125 is working well : English when page is english and german when page is german. 
I've enable localization in my search results webpart
I've added "RefinableString125" as selected property
I've customized the result template to display it : 
  _<p>School : {{RefinableString125}}</p>_ 
However, the value display is still the complete ID and not the translated value : 
_School : GP0|#eeff7dc1-c6dd-48f4-aaf8-93f4fe0fb731;L0|#0eeff7dc1-c6dd-48f4-aaf8-93f4fe0fb731|HEI - Filière Technologies du vivant;GTSet|#ac166fd5-7dc7-43f2-8bc3-68be0a3cd2e9_

**Expected behavior**
I was expecting the result display engine to translate this value to the localized term, without id, so "School : Life Technologies degree program"

**Screenshots**
![image](https://user-images.githubusercontent.com/30627006/215717291-33e1e2a3-c68b-481b-a75c-f75bd6cedac0.png)

**Desktop (please complete the following information):**
 - Browser : Edge
 - SharePoint Online Communication Site

**Additional context**
I've also tried to link RefinableString125 to a custom slot in the web and use the slot like this : 
_<p>School : {{slot item @root.slots.CustBenEcole}}</p>_
But the result is exactly the same.

**Comment by wobba** [comment]
In your template use `AutoRefinableString125`. This is documented at https://microsoft-search.github.io/pnp-modern-search/usage/search-results/data-sources/ for the *Enable localization* option.

**Reply by Nevinia210** [reply]
Hello @wobba ,
with that `Auto` trick I'm able to get specific term that i'm looking for, but is there possibility to get also parents of this term?
for example our current search returns this value:
`owstaxIdTaxValue:"GP0|#GUID;L0|#GUID| FirstValue:SecondeValue:ThirthValue ;GTSet|#GUID;GPP|#GUID;GPP|#GUID"`

if I call for `AutoowstaxIdTaxValue` i will get returned only `ThirthValue` translated

**Reply by wobba** [reply]
Have to see what happens in the translation code in _getLocalizedMetadata in SharePointSearchDataSource to see why it's failing. You can also enable the Debug template and look at the values outputted per item and see if you can see it translated or not.

**Reply by AnouckF** [reply]
Can we only use this Localization when we map a new refinable managed property? Or can we use the default propertie that is being created for a Manged Metadata field? Example AutoowsTaxIDStraatnaam from the propertie owsTaxtIDStraatnaam

**Reply by AnouckF** [reply]
I have this working with the auto and localization on.
